ED/build/install.sh

```diff
@@ -122,7 +122,7 @@ cd ${BIN}
 # Link to makefiles, includes, and shell scripts
 ln -sf ../make/*.mk ./
 ln -sf ../make/Makefile ./
-ln -sf ../make/include.mk.${PLATFORM} ./include.mk
+ln -sf ../make/include.${PLATFORM}.mk ./include.mk
```

Comment:
Can you provide rational for why you've changed the naming order/scheme?  This would change how EVERYONE has to do their compiles and so I don't think we should make a change like this without 1) clear documentation for EVERYONE (that means changing ALL of the install file names) and 2) providing a rationale for why this is better than the status quo

Reply:
I changed this because there were two include files for every platform, that is opt and dbg. Some commits ago Dan incorporated the options into the include file (see https://github.com/EDmodel/ED2/blob/master/ED/build/make/include.mk.opt.intel) but the naming convention stayed the same, resulting in two identical copies (like include.mk.opt.intel and include.mk.dbg.intel). Since the options for the optimization are inside the file I thought it would make sense to update the install.sh accordingly.
This change happened in PR #200 where 
ln -sf ../make/include.mk.${OPT}.${PLATFORM} ./include.mk
became
ln -sf ../make/include.mk.${PLATFORM} ./include.mk

This present commit only changes the naming convention putting .mk at the end in accordance to standard practice (https://www.gnu.org/software/make/manual/html_node/Include.html).

Now I understand that this is annoying because the includes have to we updated but I think it's good to avoid multiplication of files that are copies of one another. 
So I suggest that by looking at the example for intel (https://github.com/manfredo89/ED2/blob/0d08d0488269b014cf0ba5231cecb1d840d3d6a8/ED/build/make/include.intel.mk) you incorporate the compiler options inside your platform specific include. 
Also by doing this we could probably later on delete some includes that are obsolete (like the one for macOS does not work, and probably it's not the only one).
As a side note I think that most people should be able to compile with the intel platform which also provide better performance than other compilers. So do you have particular reasons not to use intel compilers?

Reply:
@manfredo89 Yes, intel compilers are not free.

Reply:
I downloaded them from the intel website for free. You just have to give an academic email address. I am just not sure whether only students qualify or postdoc/professors as well.

Reply:
According to the [intel website](https://software.intel.com/en-us/qualify-for-free-software/), free licence is only for students and professors aiming to use it in classrooms.  People actively developing ED2 may be able to qualify through the open-source contributor (not sure about this one), but in any case, not everyone using ED2 is developing the code.  As far as I know, gfortran and gcc don't have such restrictions.

Reply:
Just to reiterate what Marcos said, we should absolutely NOT assume everybody will use the same compiler or randomly delete compile scripts.  Having compiled (slightly older) versions of ED on about 7 different systems now, do not underestimate how finicky the build can be.  

Also, I wouldn't get rid of the current MacOS build script until someone gets a chance to update it or consensus is that it will no longer be supported.  In general, unilaterally removing or adding things without heads-up or opportunity for group discussion make tracking down the origins of bugs and breaks much harder.

Reply:
Just wanted to weigh in as well and support the maintaining support for multiple compilers.  My group has exclusively used gfortran / openMPI for compiling/running ED2 on several different platforms and OSs, as well as HDF5 libs.  I do not want to loose the capacity to be flexible with compilation support since, as far as I know, it does not really add a ton of overhead.  As it is now, different groups maintain their specific opt files to work on specific machines and I don't have a problem with that.  I would say that after that big make / install changes that happened about a year ago the build was working well.

Reply:
Ok. Let's get this straight then. First I am not advocating to scrap support for compilers. I am perfectly happy with people using gfortran. This commit is about avoiding the duplication of includes (opt/dbg) where most of the rest of the file is the same.

About Mac I have solved the issues and committed the changes some time ago. The problem was that the ld utility was not adding common symbols to the archive. Some background here:
https://stackoverflow.com/questions/33277964/linking-problems-for-non-explicit-interfaces
https://stackoverflow.com/questions/34595766/object-files-not-properly-added-to-archive-on-mac
However I should say that I don't know of anyone else having compiled ED on a mac (there were also issues in the code related to that, see the get_cpu function).
So since I can compile on my mac without the need for a specific include what is then the reason to keep it?

Reply:
OK, fair enough.  It looks like folks would have to update HDF and compiler options, accordingly but removing duplication is helpful if it works across machines.

Reply:
Also as a side note, beside adding lianas which is "my stuff", most of the content of the PR #200 was removing unused variable that were giving (hundreds of) warnings at compile time. When compiling with all warnings enabled the compilation fails because of some interface problems. I have already solved this issue and I am now able to compile with all checks enabled. I hope to be able to merge also this last improvements at some point.

The way I see this is as something beneficial for everyone, since for me it's really hard to debug code that gives a thousand warnings. When I started using it it took me almost a month to compile on my mac. This is also because the macosx include was not maintained. 

My view was that what is personal stays on the personal branches and what is common (and maintained) stays on the main branch. I think this is a reasonable approach to open source software. Hope this makes sense.